src/sources/http_scrape/integration_tests.rs

```diff
@@ -0,0 +1,229 @@
+        None,
+        DeserializerConfig::NativeJson,
+        default_framing_message_based(),
+        None,
+        None,
+        None,
+    ))
+    .await;
+
+    let trace = events[0].as_trace();
+    assert_eq!(trace.as_map()[log_schema().source_type_key()], NAME.into());
+}
+
+/// Passing no authentication for the auth-gated endpoint should yield errors.
+#[tokio::test]
+async fn unauthorized_no_auth() {
+    run_error(HttpScrapeConfig::new(
+        "http://dufs-auth:5000/logs/json.json".to_string(),
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::Json,
+        default_framing_message_based(),
+        None,
+        None,
+        None,
+    ))
+    .await;
+}
+
+/// Passing the incorrect credentials for the auth-gated endpoint should yield errors.
+#[tokio::test]
+async fn unauthorized_wrong_auth() {
+    run_error(HttpScrapeConfig::new(
+        "http://dufs-auth:5000/logs/json.json".to_string(),
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::Json,
+        default_framing_message_based(),
+        None,
+        None,
+        Some(Auth::Basic {
+            user: "white_rabbit".to_string(),
+            password: "morpheus".to_string(),
+        }),
+    ))
+    .await;
+}
+
+/// Passing the correct credentials for the auth-gated endpoint should succeed.
+#[tokio::test]
+async fn authorized() {
+    run_compliance(HttpScrapeConfig::new(
+        "http://dufs-auth:5000/logs/json.json".to_string(),
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::Json,
+        default_framing_message_based(),
+        None,
+        None,
+        Some(Auth::Basic {
+            user: "user".to_string(),
+            password: "pass".to_string(),
+        }),
+    ))
+    .await;
+}
+
+/// Passing the CA file for TLS should yield errors.
+#[tokio::test]
+async fn tls_invalid_ca() {
+    run_compliance(HttpScrapeConfig::new(
+        "https://dufs-https:5000/logs/json.json".to_string(),
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::Json,
+        default_framing_message_based(),
+        None,
+        Some(TlsConfig {
+            ca_file: Some(tls::TEST_PEM_INTERMEDIATE_CA_PATH.into()),
+            ..Default::default()
+        }),
+        None,
+    ))
+    .await;
+}
+
+/// Passing the correct CA file for TLS should succeed.
+#[tokio::test]
+async fn tls_valid() {
+    run_compliance(HttpScrapeConfig::new(
+        "https://dufs-https:5000/logs/json.json".to_string(),
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::Json,
+        default_framing_message_based(),
+        None,
+        Some(TlsConfig {
+            ca_file: Some(tls::TEST_PEM_CA_PATH.into()),
+            ..Default::default()
+        }),
+        None,
+    ))
+    .await;
+}
+
+/// The source should shutdown cleanly when the shutdown signal is received.
+/// TODO this can probably be extracted into the test_utils and generalized for other sources to
+/// use.
```

Comment:
There's currently a separate `shutdown` test file that contains these sorts of tests, though I'm not suggesting that's the best place for them.

Reply:
I _think_ I remember a desire to not have them there anymore - but it was probably in slack and can't be found.

https://github.com/vectordotdev/vector/blob/master/tests/shutdown.rs <- file in question

Reply:
Hummm ... The existing test I modeled this test case off of was in the source itself:

https://github.com/vectordotdev/vector/blob/c402e6ed94400959dbc298d4b342dcb0842089d4/src/sources/socket/mod.rs#L491

Reply:
I'd be happy with a central "shutdown" test if it didn't need to be amended for every components and did some magical "I test shutdown for every component."

Reply:
👍 I think having a shutdown test per component makes sense to validate that they handle the signal correctly and don't, for example, hang forever.

Reply:
I'll remove the TODO